src/blockrelay/blockrelay_common.cpp

```diff
@@ -104,23 +107,61 @@ bool ThinTypeRelay::HasBlockRelayTimerExpired(const uint256 &hash)
 
 bool ThinTypeRelay::IsBlockRelayTimerEnabled()
```

Comment:
Is there a reason this method is written out "long-hand"?  The repeated `else if` cases are all nearly identical, with the only differences being which conditions are NOT'ed, which makes it hard to review and very easy to miss logic errors.

Provided there aren't any intended special caveats I'm missing in the logic, wouldn't it be cleaner to reduce this method down similar to below?  I've revised the comments in the suggested change with my understanding of what this method is supposed to do as well as modifying the code.

NOTE: I also wrote a quick verification program that tests every possible input against the current implementation vs the suggestion below and the results deviate in 5 of 64 cases.  This is because these 5 cases fall through all checks in the original implementation when I don't believe they should.
```suggestion
bool ThinTypeRelay::IsBlockRelayTimerEnabled()
{
    // Only engage the timer if one or more, but not all, thin type relays are active.
    // If all types are active, or all inactive, then we do not need the timer.
    // Generally speaking all types will be active and we can return early.
    if (IsThinBlocksEnabled() && IsGrapheneBlockEnabled() && IsCompactBlocksEnabled())
        return false;
    if (!IsThinBlocksEnabled() && !IsGrapheneBlockEnabled() && !IsCompactBlocksEnabled())
        return false;

    // The thin relay timer is only relevant if we have a specific thin relay type active
    // AND we have peers connected which also support that thin relay type
    bool fThinBlockPossible = IsThinBlocksEnabled() && nThinBlockPeers > 0;
    bool fGraphenePossible = IsGrapheneBlockEnabled() && nGraphenePeers > 0;
    bool fCompactBlockPossible = IsCompactBlocksEnabled() && nCompactBlockPeers > 0;

    return fThinBlockPossible || fGraphenePossible || fCompactBlockPossible;
}
```

Reply:
@Justaphf thanks, will take a look...you're probably right...i think i just wrote it that way because i was writing down each case as i went along to make sure the test cases worked out.  It could probably be condensed now...

Reply:
@Justaphf great, it works...much better ....thankyou!!!  Do you want to submit a PR to my branch and I can merge it in?

Reply:
Added a PR against your branch.

Reply:
thanks, merged.